com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -191,23 +190,89 @@ public UniversalRenderer(UniversalRendererData data) : base(data)
+                enableMSAA: false,
+                name: "_CameraDepthTexture");
+            m_NormalsTexture = RTHandles.Alloc(URPShaderIDs._CameraNormalsTexture, "_CameraNormalsTexture");
+            if (renderingMode == RenderingMode.Deferred)
             {
-                m_GBufferHandles = new RenderTargetHandle[(int)DeferredLights.GBufferHandles.Count];
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.DepthAsColor].Init("_GBufferDepthAsColor");
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.Albedo].Init("_GBuffer0");
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.SpecularMetallic].Init("_GBuffer1");
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.NormalSmoothness].Init("_GBuffer2");
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.Lighting] = new RenderTargetHandle();
-                m_GBufferHandles[(int)DeferredLights.GBufferHandles.ShadowMask].Init("_GBuffer4");
+                m_GBufferHandles = new RTHandle[(int)DeferredLights.GBufferHandles.Count];
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.DepthAsColor] = RTHandles.Alloc(
+                    Vector2.one,
+                    colorFormat: DeferredLights.GetGBufferFormat(DeferredLights.GBufferHandles.DepthAsColor),
+                    name: "_CameraOpaqueTexture");
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.Albedo] = RTHandles.Alloc(
+                    Vector2.one,
+                    colorFormat: DeferredLights.GetGBufferFormat(DeferredLights.GBufferHandles.Albedo),
+                    name: "_GBuffer0");
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.SpecularMetallic] = RTHandles.Alloc(
+                    Vector2.one,
+                    colorFormat: DeferredLights.GetGBufferFormat(DeferredLights.GBufferHandles.SpecularMetallic),
+                    name: "_GBuffer1");
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.NormalSmoothness] = RTHandles.Alloc(
+                    Vector2.one,
+                    colorFormat: DeferredLights.GetGBufferFormat(DeferredLights.GBufferHandles.NormalSmoothness, accurateGbufferNormals),
+                    name: "_GBuffer2");
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.Lighting] = null; // Set to be m_ActiveCameraAttachments during runtime
+                m_GBufferHandles[(int)DeferredLights.GBufferHandles.ShadowMask] = RTHandles.Alloc(
+                    Vector2.one,
+                    colorFormat: DeferredLights.GetGBufferFormat(DeferredLights.GBufferHandles.ShadowMask),
+                    name: "_GBuffer4");
             }
-            m_OpaqueColor.Init("_CameraOpaqueTexture");
-            m_DepthInfoTexture.Init("_DepthInfoTexture");
-            m_TileDepthInfoTexture.Init("_TileDepthInfoTexture");
+            m_OpaqueColor = RTHandles.Alloc(size =>
```

Comment:
Should create upon demand.

Reply:
What's the proper strategy here? Should I use a property with get to lazy alloc such as:

```c#
private RTHandle m_OpaqueColor {
    get {
        if (m_OpaqueColor_ == null) {
            m_OpaqueColor_ = RTHandles.Alloc(...);
        }
        return m_OpaqueColor_ ;
    }
}
```

Reply:
The above can work if you still make sure to dealloc / cleanup per-camera.
  You want to avoid having resources lingering between multiple frames if they are not needed.
  
  Another approach is to delegate to the pass to alloc/dealloc as it was with GetTemRT/ReleaseTempRT

Reply:
> Another approach is to delegate to the pass to alloc/dealloc as it was with GetTemRT/ReleaseTempRT

Is the goal to Allocate and Free on each frame? This kind of goes against the dynamic resizing feature of RTHandles that do linger.